src/HttpAsyncClientDecorator.php

```diff
@@ -21,6 +23,7 @@ trait HttpAsyncClientDecorator
 
     /**
      * @see HttpAsyncClient::sendAsyncRequest
+     * @return Promise<ResponseInterface>
```

Comment:
we need to fix this in the HttpAsyncClient instead. then it will be inferred for all the various implementations here:

https://github.com/php-http/httplug/blob/625ad742c360c8ac580fcc647a1541d29e257f67/src/HttpAsyncClient.php#L20